### ISSUE TITLE
feature: Make zebra batch verification consensus code API compatible

### ISSUE THREAD

**Issue by AlvinKuruvilla** [ZcashFoundation/zebra#8569]
### Motivation

In [this](https://docs.rs/zebra-consensus/latest/src/zebra_consensus/primitives/halo2.rs.html#81) line within the docs of the zebra-consensus module, there is a note that this code is temporary till Halo2 supports batch verification. From reading the comments, it sounds like the eventual goal is to make this portion API agnostic (i.e. do not rely on orchard). From my understanding, halo2 supports batch verification, so I was wondering if that comment can be resolved now to make it API agnostic?

**Comment by mpguerra** [comment]
Thank you for reporting this! Whilst that is the goal eventually, we are prioritising other work at the moment :)

**Comment by mpguerra** [comment]
I wonder if there is an opportunity to do this at the same time as #8155

**Comment by arya2** [comment]
> I wonder if there is an opportunity to do this at the same time as #8155

Yep, the orchard batch validator currently expects action bundles where Zebra provides circuit instances and proofs, but that should be easy to change in either Zebra or orchard and it should otherwise be a drop-in replacement after de-duplicating the types.
